File: Golang/Easy/1957_Delete_Characters_to_Make_Fancy_String/solution.go
package deletecharacterstomakefancystring

import "strings"

func makeFancyString(s string) string {
	output := []string{}
	output = append(output, string(s[0]))
	count := 1

	for i := 1; i < len(s); i++ {
		c := string(s[i])
		if string(output[len(output)-1]) != c {
			output = append(output, c)
			count = 1
		} else {
			if count >= 2 {
				continue
			} else {
				count += 1
				output = append(output, c)
			}
		}
	}

	return strings.Join(output, "")
}

File: Python/Easy/1957_Delete_Characters_to_Make_Fancy_String/solution.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        output = [s[0]]
        count = 1
        for i in range(1, len(s)):
            c = s[i]
            if c != output[-1]:
                output.append(c)
                count = 1
            else:
                if count >= 2:
                    continue
                else:
                    count += 1
                    output.append(c)
        return "".join(output)

